haystack/modeling/model/multimodal/__init__.py

```diff
@@ -0,0 +1,198 @@
+        Haystack applies some default parameters to some models. They can be overridden by users by specifying the
+        desired value in this parameter. See `DEFAULT_MODEL_PARAMS`.
+    :param feature_extractor_kwargs: dictionary of parameters to pass to the feature extractor's initialization (revision, use_auth_key, etc...)
+        Haystack applies some default parameters to some models. They can be overridden by users by specifying the
+        desired value in this parameter. See `DEFAULT_MODEL_PARAMS`.
+    :param pooler_kwargs: dictionary of parameters to pass to the pooler's initialization (summary_last_dropout, summary_activation, etc...)
+        Haystack applies some default parameters to some models. They can be overridden by users by specifying the
+        desired value in this parameter. See `POOLER_PARAMETERS`.
+    """
+    autoconfig_kwargs = autoconfig_kwargs or {}
+    model_kwargs = model_kwargs or {}
+    feature_extractor_kwargs = feature_extractor_kwargs or {}
+    pooler_kwargs = pooler_kwargs or {}
+
+    if not pretrained_model_name_or_path or not isinstance(pretrained_model_name_or_path, (str, Path)):
+        raise ValueError(
+            f"{pretrained_model_name_or_path} is not a valid 'pretrained_model_name_or_path' value. "
+            "Please provide a string or a Path object."
+        )
+    model_name = str(pretrained_model_name_or_path)
+    model_type = ""
+    model_wrapper_class: Type[HaystackModel]
+
+    # Prepare the kwargs the model wrapper expects (see each wrapper's init for details)
+    wrapper_kwarg_groups = {}
+    wrapper_kwarg_groups["model_kwargs"] = model_kwargs
+
+    # SentenceTransformers are much faster, so use them whenever possible
+    if _is_sentence_transformers_model(
+        pretrained_model_name_or_path, use_auth_token=autoconfig_kwargs.get("use_auth_token", False)
+    ):
+        model_wrapper_class = HaystackSentenceTransformerModel
+        try:
+            # Use AutoConfig to log some more info about the model class
+            config = AutoConfig.from_pretrained(pretrained_model_name_or_path=model_name, **autoconfig_kwargs)
+            model_type = config.model_type
+        except Exception as e:
+            logger.debug(f"Can't find model type for {pretrained_model_name_or_path}: {e}")
+
+        if feature_extractor_kwargs is not None:
+            logger.debug(
+                "Can't forward feature_extractor_kwargs to a SentenceTransformers model. "
+                "These kwargs are being dropped. "
+                f"Content of feature_extractor_kwargs: {feature_extractor_kwargs}"
+            )
+
+    else:
+        # Use AutoConfig to understand the model class
+        config = AutoConfig.from_pretrained(pretrained_model_name_or_path=model_name, **autoconfig_kwargs)
+        if not config.model_type:
+            logger.error(
+                f"Model type not understood for '{pretrained_model_name_or_path}'. Please provide the name of "
```

Comment:
Ok! How about:
```suggestion
                f"Unknown model type detected for '{pretrained_model_name_or_path}'. You need to provide the identifier of "
```

Reply:
How about we tell them explicitly:
"Haystack cannot handle the model type for blahblah. Provide the identifier of a model that Haystack can download from the Hugging Face Hub, otherwise your system may crash."

Reply:
Yes we can, but I think the current one is more specific. It's not simply that Haystack cannot handle this model: it cannot because it can't recognize/parse/understand the model type. If the user owns the model, this error can be fixed by changing (or adding if it's missing) the model type in the Hub.